In [103]:
import pandas as pd

In [104]:
df = pd.read_csv('Tables/lwerst2019.csv', sep=';', usecols=['Nr', 'Wahlkreis', 'Name', 'Wahlb. insges.', 'Wähler'])
df.head()

Nr  Wahlkreis                                      Name  Wahlb. insges.  \
0  1306         19     Bornstedt - Hermann-Mattern-Promenade            1050   
1  1601         19                                     Grube             345   
2  8401         19  Fahrland - V.-Stechow-Str./Rotkehlchenwe             809   
3  8501         19                              Neu Fahrland            1281   
4  8601         19      Groß Glienicke - Glienicker Dorfstr.             670   

   Wähler  
0     696  
1     210  
2     421  
3     671  
4     316

In [105]:
#Filterraus Briefwahlbezirk

substring = 'Briefwahlbezirk'
filter = df['Name'].str.contains(substring)
filtered_df = df[~filter]
filtered_df

Nr  Wahlkreis                                      Name  \
0    1306         19     Bornstedt - Hermann-Mattern-Promenade   
1    1601         19                                     Grube   
2    8401         19  Fahrland - V.-Stechow-Str./Rotkehlchenwe   
3    8501         19                              Neu Fahrland   
4    8601         19      Groß Glienicke - Glienicker Dorfstr.   
..    ...        ...                                       ...   
211  7104         22              Stern - In der Aue/Gluckstr.   
212  7111         22            Stern - Jagdhausstr./Eulenkamp   
213  7203         22   Drewitz - Paul-Wegener-Str./Priesterweg   
214  7206         22  Drewitz - Gerlachstr/E-v-Winterstein-Str   
215  7304         22   Kirchsteigfeld - Pierre-de-Gayette-Str.   

     Wahlb. insges.  Wähler  
0              1050     696  
1               345     210  
2               809     421  
3              1281     671  
4               670     316  
..              ...     ...  
211            1285     669  
212            1035     499  
213             731     271  
214             711     297  
215             648     309  

[166 rows x 5 columns]

In [106]:
# Add Wahlbeteiligung

filtered_df['Wahlbeteiligung'] = filtered_df[['Wähler']].div(filtered_df['Wahlb. insges.'], axis=0)
filtered_df['Wahlbeteiligung'] = filtered_df['Wahlbeteiligung'] *100
filtered_df['Wahlbeteiligung'] = filtered_df['Wahlbeteiligung'].astype(float).round(2)


filtered_df.head()

/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/391416788.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df['Wahlbeteiligung'] = filtered_df[['Wähler']].div(filtered_df['Wahlb. insges.'], axis=0)
/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/391416788.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df['Wahlbeteiligung'] = filtered_df['Wahlbeteiligung'] *100
/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/391416788.py:5: SettingWith

Nr  Wahlkreis                                      Name  Wahlb. insges.  \
0  1306         19     Bornstedt - Hermann-Mattern-Promenade            1050   
1  1601         19                                     Grube             345   
2  8401         19  Fahrland - V.-Stechow-Str./Rotkehlchenwe             809   
3  8501         19                              Neu Fahrland            1281   
4  8601         19      Groß Glienicke - Glienicker Dorfstr.             670   

   Wähler  Wahlbeteiligung  
0     696            66.29  
1     210            60.87  
2     421            52.04  
3     671            52.38  
4     316            47.16

In [107]:
# repeat for Kommunalwahlen 2024

df1 = pd.read_csv('Tables/kw2024.csv', sep=';', usecols=['Gebietsart', 'Gebietsname', 'Gebietsnummer', 'Bezirksnummer', 'Wahlberechtigte gesamt (A)', 'Waehler gesamt (B)'])
filter = df1['Gebietsname'].str.contains(substring)
filtered_df1 = df1[~filter]

#Wahlbeteiligung hinzufuegen
filtered_df1['Wahlbeteiligung_kw2024'] = filtered_df1[['Waehler gesamt (B)']].div(filtered_df1['Wahlberechtigte gesamt (A)'], axis=0)
filtered_df1['Wahlbeteiligung_kw2024'] = filtered_df1['Wahlbeteiligung_kw2024'] *100
filtered_df1['Wahlbeteiligung_kw2024'] = filtered_df1['Wahlbeteiligung_kw2024'].astype(float).round(2)

filtered_df1.head()


/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/509594793.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df1['Wahlbeteiligung_kw2024'] = filtered_df1[['Waehler gesamt (B)']].div(filtered_df1['Wahlberechtigte gesamt (A)'], axis=0)
/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/509594793.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df1['Wahlbeteiligung_kw2024'] = filtered_df1['Wahlbeteiligung_kw2024'] *100
/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0

Gebietsnummer   Gebietsart                Gebietsname  Bezirksnummer  \
0  1205400000002102  STIMMBEZIRK        Nauener Vorstadt 02         2102.0   
1  1205400000004101  STIMMBEZIRK  Historische Innenstadt 01         4101.0   
2  1205400000004104  STIMMBEZIRK  Historische Innenstadt 04         4104.0   
3  1205400000004106  STIMMBEZIRK  Historische Innenstadt 06         4106.0   
4  1205400000004109  STIMMBEZIRK  Historische Innenstadt 09         4109.0   

   Wahlberechtigte gesamt (A)  Waehler gesamt (B)  Wahlbeteiligung_kw2024  
0                        1042                 484                   46.45  
1                        1202                 593                   49.33  
2                         988                 462                   46.76  
3                         939                 447                   47.60  
4                        1247                 574                   46.03

In [108]:
# repeat for Europawahlen 2024

df2 = pd.read_csv('Tables/euro2024.csv', sep=';', usecols=['Gebietsart', 'Gebietsname', 'Bezirksnummer', 'Wahlberechtigte gesamt (A)', 'Waehler gesamt (B)'])
filter = df2['Gebietsart'].str.contains("BRIEFWAHLBEZIRK")
filtered_df2 = df2[~filter]

#Wahlbeteiligung hinzufuegen
filtered_df2['Wahlbeteiligung_euro2024'] = filtered_df2[['Waehler gesamt (B)']].div(filtered_df2['Wahlberechtigte gesamt (A)'], axis=0)
filtered_df2['Wahlbeteiligung_euro2024'] = filtered_df2['Wahlbeteiligung_euro2024'] *100
filtered_df2['Wahlbeteiligung_euro2024'] = filtered_df2['Wahlbeteiligung_euro2024'].astype(float).round(2)

filtered_df2

/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/4148420935.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df2['Wahlbeteiligung_euro2024'] = filtered_df2[['Waehler gesamt (B)']].div(filtered_df2['Wahlberechtigte gesamt (A)'], axis=0)
/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/4148420935.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df2['Wahlbeteiligung_euro2024'] = filtered_df2['Wahlbeteiligung_euro2024'] *100
/var/folders/kl/gsxq5z0j7pg9svkfb

Gebietsart   Gebietsname  Bezirksnummer  Wahlberechtigte gesamt (A)  \
0    STIMMBEZIRK     Bornim 01         1101.0                        1146   
1    STIMMBEZIRK  Bornstedt 03         1303.0                        1196   
2    STIMMBEZIRK  Bornstedt 04         1304.0                        1283   
3    STIMMBEZIRK  Bornstedt 07         1307.0                        1204   
4    STIMMBEZIRK     Bornim 02         1102.0                        1453   
..           ...           ...            ...                         ...   
169  STIMMBEZIRK    Drewitz 03         7203.0                         783   
170  STIMMBEZIRK    Drewitz 04         7204.0                         760   
171  STIMMBEZIRK  Marquardt 01         8201.0                         962   
172  STIMMBEZIRK   Fahrland 01         8401.0                         846   
173  STIMMBEZIRK   Fahrland 02         8402.0                        1114   

     Waehler gesamt (B)  Wahlbeteiligung_euro2024  
0                   606                     52.88  
1                   604                     50.50  
2                   712                     55.49  
3                   569                     47.26  
4                   812                     55.88  
..                  ...                       ...  
169                 339                     43.30  
170                 326                     42.89  
171                 569                     59.15  
172                 441                     52.13  
173                 572                     51.35  

[134 rows x 6 columns]

In [109]:
# repeat for Europwahl 2019
df3 = pd.read_csv('Tables/ew2019.csv', sep=';', usecols=['Nr', 'Name', 'Wahlb. insges.', 'Wähler'])

# filter out Briefwahlbezirk

substring = 'Briefwahlbezirk'
filter = df3['Name'].str.contains(substring)
filtered_df3 = df3[~filter]

# Add Wahlbeteiligung

filtered_df3['Wahlbeteiligung_euro2019'] = filtered_df3[['Wähler']].div(filtered_df3['Wahlb. insges.'], axis=0)
filtered_df3['Wahlbeteiligung_euro2019'] = filtered_df3['Wahlbeteiligung_euro2019'] *100
filtered_df3['Wahlbeteiligung_euro2019'] = filtered_df3['Wahlbeteiligung_euro2019'].astype(float).round(2)



/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/2676869739.py:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df3['Wahlbeteiligung_euro2019'] = filtered_df3[['Wähler']].div(filtered_df3['Wahlb. insges.'], axis=0)
/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/2676869739.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df3['Wahlbeteiligung_euro2019'] = filtered_df3['Wahlbeteiligung_euro2019'] *100
/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipyker

In [110]:
# repeat for Kommunalwahl 2019
df4 = pd.read_csv('Tables/kw2019.csv', sep=';', usecols=['Nr', 'Name', 'Wahlb. insges.', 'Wähler'])

# filter out Briefwahlbezirk

substring = 'Briefwahlbezirk'
filter = df4['Name'].str.contains(substring)
filtered_df4 = df4[~filter]

# Add Wahlbeteiligung

filtered_df4['Wahlbeteiligung_kw2019'] = filtered_df4[['Wähler']].div(filtered_df4['Wahlb. insges.'], axis=0)
filtered_df4['Wahlbeteiligung_kw2019'] = filtered_df4['Wahlbeteiligung_kw2019'] *100
filtered_df4['Wahlbeteiligung_kw2019'] = filtered_df4['Wahlbeteiligung_kw2019'].astype(float).round(2)
filtered_df4


/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/3515783023.py:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df4['Wahlbeteiligung_kw2019'] = filtered_df4[['Wähler']].div(filtered_df4['Wahlb. insges.'], axis=0)
/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/3515783023.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df4['Wahlbeteiligung_kw2019'] = filtered_df4['Wahlbeteiligung_kw2019'] *100
/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_96

Nr                                      Name  Wahlb. insges.  Wähler  \
0    2102           Nauener Vorstadt - Neuer Garten            1050     512   
1    2103      Nauener Vorstadt - Russische Kolonie             953     520   
2    2104       Nauener Vorstadt - Rote Kaserne/SAP            1044     533   
3    4103    Innenstadt - Hans-Thoma-Str./Türckstr.            1160     495   
4    4105                    Innenstadt - Yorckstr.            1265     632   
..    ...                                       ...             ...     ...   
171  7204               Drewitz - Conrad-Veidt-Str.             842     280   
172  7205            Drewitz - Willy-A.-Kleinau-Weg             758     224   
173  7206  Drewitz - Gerlachstr/E-v-Winterstein-Str             761     227   
174  7301  Kirchsteigfeld - Eleonore-Prochaska-Str.            1039     440   
175  7303   Kirchsteigfeld - Zum Teich/Schinkelstr.            1138     587   

     Wahlbeteiligung_kw2019  
0                     48.76  
1                     54.56  
2                     51.05  
3                     42.67  
4                     49.96  
..                      ...  
171                   33.25  
172                   29.55  
173                   29.83  
174                   42.35  
175                   51.58  

[132 rows x 5 columns]

In [111]:
# repeat for Bundestagswahl 2021

df5 = pd.read_csv('Tables/bw2021.csv', sep=';', usecols=['Gebietsnummer', 'Wahlberechtigte gesamt (A)', 'Gebietsart', 'Waehler gesamt (B)'])

# filter out Briefwahlbezirk

substring = 'BRIEFWAHLBEZIRK'
filter = df5['Gebietsart'].str.contains(substring)
filtered_df5 = df5[~filter]

# Add Wahlbeteiligung

filtered_df5['Wahlbeteiligung_bw2021'] = filtered_df5[['Waehler gesamt (B)']].div(filtered_df5['Wahlberechtigte gesamt (A)'], axis=0)
filtered_df5['Wahlbeteiligung_bw2021'] = filtered_df5['Wahlbeteiligung_bw2021'] *100
filtered_df5['Wahlbeteiligung_bw2021'] = filtered_df5['Wahlbeteiligung_bw2021'].astype(float).round(2)
filtered_df5


/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/1925881336.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df5['Wahlbeteiligung_bw2021'] = filtered_df5[['Waehler gesamt (B)']].div(filtered_df5['Wahlberechtigte gesamt (A)'], axis=0)
/var/folders/kl/gsxq5z0j7pg9svkfbwhgsx8h0000gn/T/ipykernel_9650/1925881336.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df5['Wahlbeteiligung_bw2021'] = filtered_df5['Wahlbeteiligung_bw2021'] *100
/var/folders/kl/gsxq5z0j7pg9svkfbwhgs

Gebietsnummer   Gebietsart  Wahlberechtigte gesamt (A)  \
0    1205400000001101  STIMMBEZIRK                        1113   
1    1205400000001102  STIMMBEZIRK                        1353   
2    1205400000001301  STIMMBEZIRK                        1082   
3    1205400000001302  STIMMBEZIRK                        1136   
4    1205400000001303  STIMMBEZIRK                        1150   
..                ...          ...                         ...   
324  1207202402400015  STIMMBEZIRK                        1605   
325  1207202402400019  STIMMBEZIRK                        1175   
326  1207202402400022  STIMMBEZIRK                         463   
327  1207202402400025  STIMMBEZIRK                         373   
328  1207202402400026  STIMMBEZIRK                         280   

     Waehler gesamt (B)  Wahlbeteiligung_bw2021  
0                   561                   50.40  
1                   692                   51.15  
2                   498                   46.03  
3                   551                   48.50  
4                   585                   50.87  
..                  ...                     ...  
324                 733                   45.67  
325                 453                   38.55  
326                 294                   63.50  
327                 224                   60.05  
328                 180                   64.29  

[220 rows x 5 columns]

In [112]:
# Prepare Tables for merge
# rename to distinct

lwerst2019_df = filtered_df
kw2024_df = filtered_df1
euro2024_df = filtered_df2
euro2019_df = filtered_df3
kw2019_df = filtered_df4
bw2021_df = filtered_df5

# drop unnecessary columns
kw2024_df = kw2024_df.drop(columns=['Gebietsart', 'Gebietsname', 'Wahlberechtigte gesamt (A)', 'Waehler gesamt (B)'])
lwerst2019_df = lwerst2019_df.drop(columns=['Wahlkreis', 'Wahlb. insges.', 'Wähler'])
lwerst2019_df = lwerst2019_df.rename(columns={"Nr" : "Bezirksnummer", "Wahlbeteiligung" : "Wahlbeteiligung_lwerst2019"})
euro2024_df = euro2024_df.drop(columns=['Gebietsart', 'Gebietsname', 'Wahlberechtigte gesamt (A)', 'Waehler gesamt (B)'])
euro2019_df = euro2019_df.drop(columns=['Name', 'Wahlb. insges.', 'Wähler'])
euro2019_df = euro2019_df.rename(columns={"Nr" : "Bezirksnummer"})
kw2019_df = kw2019_df.rename(columns={"Nr" : "Bezirksnummer"})
kw2019_df = kw2019_df.drop(columns=['Name', 'Wahlb. insges.', 'Wähler'])
bw2021_df = bw2021_df.drop(columns=['Gebietsart', 'Wahlberechtigte gesamt (A)', 'Waehler gesamt (B)'])

#combined_df

combined_df = pd.merge(bw2021_df, kw2024_df, on="Gebietsnummer")
combined_df = combined_df.drop(columns=["Gebietsnummer"])
combined_df = pd.merge(combined_df, euro2019_df, on="Bezirksnummer")
combined_df = pd.merge(combined_df, euro2024_df, on="Bezirksnummer")
combined_df = pd.merge(combined_df, lwerst2019_df, on="Bezirksnummer")
combined_df = pd.merge(combined_df, kw2019_df, on="Bezirksnummer")

# euro2019, lw2019, kw2019, bw2021, kw2024, euro2024

combined_df = combined_df.loc[:,['Bezirksnummer','Name','Wahlbeteiligung_kw2019', 'Wahlbeteiligung_euro2019', 'Wahlbeteiligung_lwerst2019', 'Wahlbeteiligung_bw2021', 'Wahlbeteiligung_kw2024', 'Wahlbeteiligung_euro2024']]



In [113]:
#chanage index, transpose and export
combined_df_index = combined_df.set_index("Name") 
wahlbeteiligung_potsdam = combined_df_index.drop(columns=["Bezirksnummer"]).T
#combined_df_flip.drop(columns=["Bezirksnummer"]).T
#combined_df_flip.drop(columns=["Bezirksnummer"]).T.to_csv("WahlbeteiligungPotsdam.csv")

wahlbeteiligung_potsdam

Name                        Bornim - Max-Eyth-Allee/Lerchensteig  \
Wahlbeteiligung_kw2019                                     44.27   
Wahlbeteiligung_euro2019                                   44.66   
Wahlbeteiligung_lwerst2019                                 50.98   
Wahlbeteiligung_bw2021                                     50.40   
Wahlbeteiligung_kw2024                                     52.64   
Wahlbeteiligung_euro2024                                   52.88   

Name                        Bornim - Ortslage/Hügelweg  \
Wahlbeteiligung_kw2019                           52.95   
Wahlbeteiligung_euro2019                         54.15   
Wahlbeteiligung_lwerst2019                       58.15   
Wahlbeteiligung_bw2021                           51.15   
Wahlbeteiligung_kw2024                           54.36   
Wahlbeteiligung_euro2024                         55.88   

Name                        Nedlitz, Bornstedt - Viereckremise  \
Wahlbeteiligung_kw2019                                   49.04   
Wahlbeteiligung_euro2019                                 49.80   
Wahlbeteiligung_lwerst2019                               53.08   
Wahlbeteiligung_bw2021                                   46.03   
Wahlbeteiligung_kw2024                                   52.32   
Wahlbeteiligung_euro2024                                 53.70   

Name                        Bornstedt - Fliederweg  \
Wahlbeteiligung_kw2019                       52.69   
Wahlbeteiligung_euro2019                     52.39   
Wahlbeteiligung_lwerst2019                   53.68   
Wahlbeteiligung_bw2021                       48.50   
Wahlbeteiligung_kw2024                       55.84   
Wahlbeteiligung_euro2024                     55.93   

Name                        Bornstedt - Am Krongut/Katharinenholzstr  \
Wahlbeteiligung_kw2019                                         49.45   
Wahlbeteiligung_euro2019                                       50.13   
Wahlbeteiligung_lwerst2019                                     53.65   
Wahlbeteiligung_bw2021                                         50.87   
Wahlbeteiligung_kw2024                                         49.79   
Wahlbeteiligung_euro2024                                       50.50   

Name                        Bornstedt - Kirschallee/Thaerstr.  \
Wahlbeteiligung_kw2019                                  49.49   
Wahlbeteiligung_euro2019                                49.21   
Wahlbeteiligung_lwerst2019                              55.36   
Wahlbeteiligung_bw2021                                  46.23   
Wahlbeteiligung_kw2024                                  55.03   
Wahlbeteiligung_euro2024                                55.49   

Name                        Bornstedt - Hermann-Mattern-Promenade  \
Wahlbeteiligung_kw2019                                      61.30   
Wahlbeteiligung_euro2019                                    61.45   
Wahlbeteiligung_lwerst2019                                  66.29   
Wahlbeteiligung_bw2021                                      55.21   
Wahlbeteiligung_kw2024                                      61.06   
Wahlbeteiligung_euro2024                                    61.31   

Name                        Bornstedt - A.-Wolff-Platz/E.-Arendt-Str  \
Wahlbeteiligung_kw2019                                         47.46   
Wahlbeteiligung_euro2019                                       50.12   
Wahlbeteiligung_lwerst2019                                     48.62   
Wahlbeteiligung_bw2021                                         44.66   
Wahlbeteiligung_kw2024                                         46.22   
Wahlbeteiligung_euro2024                                       47.68   

Name                        Eiche - Altes Rad Ost  \
Wahlbeteiligung_kw2019                      49.47   
Wahlbeteiligung_euro2019                    49.77   
Wahlbeteiligung_lwerst2019                  53.86   
Wahlbeteiligung_bw2021                      45.58   
Wahlbeteiligung_kw2024                      53.13   
Wahlbe

In [114]:
#get certain values for the visualization

#combined_df.max()
#overall high: 66,29

#combined_df.min()
#overall low = 29.55

#combined_df_flip[["Wahlbeteiligung_kw2019", "Wahlbeteiligung_euro2019", "Wahlbeteiligung_lwerst2019", "Wahlbeteiligung_bw2021", "Wahlbeteiligung_kw2024", "Wahlbeteiligung_euro2024"]].mean().astype(float).round(2)
#mean over all columns: 48,66

combined_df_index["mean"] = wahlbeteiligung_potsdam.mean().round(2)
combined_df_index = combined_df_index.sort_values("mean")
#combined_df_index

wahlbeteiligung_potsdam = combined_df_index.drop(columns=["Bezirksnummer", "mean"]).T

wahlbeteiligung_potsdam.to_csv("WahlbeteiligungPotsdamVergleich.csv")
wahlbeteiligung_potsdam

Name                        Schlaatz - H.-Muthesius-Str./Binsenhof  \
Wahlbeteiligung_kw2019                                       31.75   
Wahlbeteiligung_euro2019                                     33.09   
Wahlbeteiligung_lwerst2019                                   35.75   
Wahlbeteiligung_bw2021                                       38.52   
Wahlbeteiligung_kw2024                                       34.38   
Wahlbeteiligung_euro2024                                     36.00   

Name                        Stern - Otto-Hahn-Ring  \
Wahlbeteiligung_kw2019                       30.75   
Wahlbeteiligung_euro2019                     31.91   
Wahlbeteiligung_lwerst2019                   36.31   
Wahlbeteiligung_bw2021                       37.49   
Wahlbeteiligung_kw2024                       36.60   
Wahlbeteiligung_euro2024                     38.30   

Name                        Schlaatz - Inselhof/Bisamkiez  \
Wahlbeteiligung_kw2019                              30.95   
Wahlbeteiligung_euro2019                            32.77   
Wahlbeteiligung_lwerst2019                          32.06   
Wahlbeteiligung_bw2021                              42.23   
Wahlbeteiligung_kw2024                              38.33   
Wahlbeteiligung_euro2024                            41.50   

Name                        Drewitz - Willy-A.-Kleinau-Weg  \
Wahlbeteiligung_kw2019                               29.55   
Wahlbeteiligung_euro2019                             30.44   
Wahlbeteiligung_lwerst2019                           35.01   
Wahlbeteiligung_bw2021                               44.77   
Wahlbeteiligung_kw2024                               39.05   
Wahlbeteiligung_euro2024                             42.07   

Name                        Stern - Leibnizring  Schlaatz - Erlenhof  \
Wahlbeteiligung_kw2019                    30.96                31.68   
Wahlbeteiligung_euro2019                  31.65                34.35   
Wahlbeteiligung_lwerst2019                37.28                37.03   
Wahlbeteiligung_bw2021                    45.69                43.25   
Wahlbeteiligung_kw2024                    40.61                38.85   
Wahlbeteiligung_euro2024                  41.82                43.02   

Name                        Drewitz - Paul-Wegener-Str./Priesterweg  \
Wahlbeteiligung_kw2019                                        30.69   
Wahlbeteiligung_euro2019                                      32.32   
Wahlbeteiligung_lwerst2019                                    37.07   
Wahlbeteiligung_bw2021                                        46.68   
Wahlbeteiligung_kw2024                                        40.81   
Wahlbeteiligung_euro2024                                      43.30   

Name                        Drewitz - Sterncenter/Erich-Pommer-Str.  \
Wahlbeteiligung_kw2019                                        37.90   
Wahlbeteiligung_euro2019                                      38.57   
Wahlbeteiligung_lwerst2019                                    41.48   
Wahlbeteiligung_bw2021                                        35.03   
Wahlbeteiligung_kw2024                                        40.43   
Wahlbeteiligung_euro2024                                      42.58   

Name                        Waldstadt II - Kiefernring/Sonnentaustr.  \
Wahlbeteiligung_kw2019                                         37.17   
Wahlbeteiligung_euro2019                                       37.91   
Wahlbeteiligung_lwerst2019                                     39.16   
Wahlbeteiligung_bw2021                                         40.56   
Wahlbeteiligung_kw2024                                         41.63   
Wahlbeteiligung_euro2024                                       42.84   

Name                        Schlaatz - Sperberhorst  ...  \
Wahlbeteiligung_kw2019                        33.99  ...   
Wahlbeteiligung_euro2019                      35.99  ...   
Wahlbeteiligung_lwerst2019                    40.28  ...   
Wahlbeteiligung_bw202

Wahlbeteiligung Durchschnitt pro Wahl

Wahlbeteiligung_kw2019        46.70
Wahlbeteiligung_euro2019      47.50
Wahlbeteiligung_lwerst2019    50.53
Wahlbeteiligung_bw2021        47.22
Wahlbeteiligung_kw2024        49.47
Wahlbeteiligung_euro2024      50.54